src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -135,7 +135,15 @@ public void End(ProduceEndType endType)
 
         private void ScheduleWrite()
         {
-            _thread.Post(_this => _this.WriteAllPending(), this);
+            // Don't post write to closed socket
+            if (!_socket.IsClosed)
```

Comment:
Is this meant to be a perf optimization? `_thread.Post` shouldn't fail because the socket is closed.

We only actually use `_socket` in the `WriteContext` methods where `_socket.IsClosed` is always checked first thing.